### ISSUE TITLE
Generate floor tiles

### ISSUE THREAD

**Issue by Kazuo256** [uspgamedev/backdoor#125]
+ [x] Place first tiles at random
+ [x] Grow regions with many dirt tiles
+ [x] Shrink regions with not enough dirt tiles

**Comment by Kazuo256** [comment]
As I tried to tackle this issue after a long time, I came to realize that keeping border tiles in the "model" map is really cumbersome. Just by addin a new floor tile, the number of tile values has mostly doubled. I believe detecting and presenting border tiles should be a responsability of the "view" map, making the "model" map a flat simple matrix, which is much easier to work with and apply procedural stuff on.

Thus, I'm thinking of moving the pattern-matching code out of the map generation code and into the "view" map code. That would solve 50% of this issue, since as it is dirt tiles are already being generated, but without borders. What I mean is, if we decide to move border tiles to the "view" map, then this issue is already done.
